docs/conventions/components.md

```diff
@@ -0,0 +1,62 @@
+
+The following sections touch upon the important aspects of maintaining a component package.
+
+## Manifest file package.json
+
+As mentioned earlier in the introduction, the `package.json` is the first and most important element to attend to when working with components' packages.
+
+Please ensure the presence of the following attributes:
+
+- `name`: name of the package. Follow naming conventions presented in the following section
+- `style`: path to a CSS file. This is the main bundled stylesheet (dist/[name].css)
+- `sass`: path to a SCSS file. Higher priority than `style`
+- `main`: path to a JavaScript file. Used by non ES6-aware tools (UMD) (dist/[name].js)
+- `module`: path to a JavaScript file file. Used by ES6-aware tools like webpack
+- `dependencies`: list of other packages' code which is required for the given package
+
+It's important to have packages within `src/systems/ec/implementations/react` folder with `"private": true` flag because these are used only for internal purposes of demonstrating implementaiton of vanilla components on ECL website.
+
+**React components should not be shared publicly. They are not an official implementation of ECL.**
+
+## Naming
+
+- `@ecl/` namespace should be used consistently throughout all packages in order to be published correctly on [ECL v2 npm organisation](https://www.npmjs.com/org/ecl).
+- The following pattern is to be followed in order to take into account systems specificity: `[ec|eu]-[implementation]-[component|layout|utility]-element`.
+
+Examples:
+
+- `@ecl/ec-component-menu` - vanilla (base) component
+- `@ecl/ec-react-component-menu` react implementation for website
+
+## Implementations
+
+Please follow along existing packages in `src/systems/ec/implementations/vanilla/packages` and `src/systems/ec/implementations/react/components`.
+
+If you have difficulties figuring out what should be where, here are a few rules of thumb:
+
+- Vanilla code which does not target any particular framework or platform goes into `vanilla/packages`. This is the first-level implementation.
+- Other implementations are done in a folder up `src/systems/ec/implementations` and `react` is currently used for facilitating components demonstration on ECL website developed in `src/website`.
+- There should be no "leaking" of rules between components, each component comes with its own set of rules and logic in isolation.
+
+## SCSS
+
+Please refer to the [dedicated conventions section regarding SCSS](./scss.md).
+
+## JavaScript
+
+Please refer to the [dedicated conventions section regarding JavaScript](./javascript.md).
+
+## Links
+
+If a component contains links, they should always lead to an internal example page instead of blank link (`#`) or external links: `../../example.html#{component_name}`
```

Comment:
that's not very clear which links we are talking about.
I guess that's for the demonstration of the component, right? If so, it may be part of a section dedicated to "specs" files